avm/res/document-db/mongo-cluster/config-firewall-rule/main.bicep

```diff
@@ -0,0 +1,84 @@
+metadata name = 'DocumentDB Mongo Clusters Config FireWall Rules'
+metadata description = 'This module config firewall rules for DocumentDB Mongo Cluster.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Optional. Whether to allow all IPs or not. Warning: No IP addresses will be blocked and any host on the Internet can access the coordinator in this server group. It is strongly recommended to use this rule only temporarily and only on test clusters that do not contain sensitive data.')
+param allowAllIPsFirewall bool = false
+
+@description('Optional. Whether to allow Azure internal IPs or not.')
+param allowAzureIPsFirewall bool = false
+
+@description('Optional. IP addresses to allow access to the cluster from.')
+param allowedSingleIPs array = []
+
+@description('Conditional. The name of the parent mongo cluster. Required if the template is used in a standalone deployment.')
+param mongoClusterName string
+
+resource mongoCluster 'Microsoft.DocumentDB/mongoClusters@2024-02-15-preview' existing = {
+  name: mongoClusterName
+}
+
+resource firewall_all 'Microsoft.DocumentDB/mongoClusters/firewallRules@2024-02-15-preview' = if (allowAllIPsFirewall) {
+  name: 'allow-all-IPs'
+  parent: mongoCluster
+  properties: {
+    startIpAddress: '0.0.0.0'
+    endIpAddress: '255.255.255.255'
+  }
+}
+
+resource firewall_azure 'Microsoft.DocumentDB/mongoClusters/firewallRules@2024-02-15-preview' = if (allowAzureIPsFirewall) {
+  name: 'allow-all-azure-internal-IPs'
+  parent: mongoCluster
+  properties: {
+    startIpAddress: '0.0.0.0'
+    endIpAddress: '0.0.0.0'
+  }
+}
+
+resource firewall_single 'Microsoft.DocumentDB/mongoClusters/firewallRules@2024-02-15-preview' = [
```

Comment:
I only now see that this is a loop. Same as in my other comment, I'd strongly suggest to change the way this module is implemented. It's not aligned with the AVM spec [Single Resource Only](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-rmfr1---category-composition---single-resource-only) which is one of the mandatory ones.

How about generalizing this module to accept a start & end IP as per the resource type, and move all the logic / cases you implemented into the parent, that is 'mongo-cluster' where you could pass the intended parameters down to this module as defined. I'd still argue they should only be examples to make it easy for you, but you can of course implement it in a different way.